Docs for new k8s configuration options (#11390)
dagster-io/dagster: docs/content/deployment/guides/kubernetes/customizing-your-deployment.mdx

This section covers common ways to customize your Dagster Helm deployment.

## Specifying custom Kubernetes configuration

> Dagster allows you to pass custom configuration through to the Kubernetes Jobs and Pods created by Dagster during execution.

Reviewer: ```suggestion
Dagster allows you to pass custom configuration to the Kubernetes Jobs and Pods created by Dagster during execution.
```

I took `through` out because it didn't feel necessary, but feel free to add it back if it does convey some special meaning

- `jobMetadata`: The Job's [Metadata](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#objectmeta-v1-meta).

The values for each of these keys is a dictionary with the YAML configuration for the underlying Kubernetes object. The Kubernetes object fields can be configured using either snake case (for example, `volume_mounts`) or camel case (`volumeMounts`). For example:

```yaml file=/deploying/kubernetes/run_k8s_config.yaml
runLauncher:
  type: K8sRunLauncher
  config:
    k8sRunLauncher:
      runK8sConfig:
        containerConfig: # raw config for the pod's main container
          resources:
            cpu: 100m
            memory: 128Mi
        podTemplateSpecMetadata: # raw config for the pod's metadata
          annotations:
            mykey: myvalue
        podSpecConfig: # raw config for the spec of the launched's pod
          nodeSelector:
            disktype: ssd
        jobSpecConfig: # raw config for the kubernetes job's spec
          ttlSecondsAfterFinished: 7200
        jobMetadata: # raw config for the kubernetes job's metadata
          annotations:
            mykey: myvalue
```

> ### Per-Job or per-Op Kubernetes Configuration

Reviewer: ```suggestion
### Per-job or per-op Kubernetes configuration
```

I know this wasn't your change, but if you don't mind while we're here

> The `dagster-k8s/config` tag allows you to pass custom configuration through to the Kubernetes Jobs and Pods created by Dagster for specific jobs or ops.

Reviewer: ```suggestion
The `dagster-k8s/config` tag allows you to pass custom configuration to the Kubernetes Jobs and Pods created by Dagster for specific jobs or ops.
```

Non-k8s run launchers and executors will ignore the `dagster-k8s/config` tag.

> If a Kubernetes configuration dictionary (like 'container_config') is specified at both the instance level in the Helm chart, and in a specific Dagster job or op, the two dictionaries will be shallowly merged, with the more specific configuration taking precedence if the same key is set in both dictionaries.

Reviewer: ```suggestion
If a Kubernetes configuration dictionary (like `container_config`) is specified at both the instance level in the Helm chart and in a specific Dagster job or op, the two dictionaries will be shallowly merged. The more specific configuration takes precedence if the same key is set in both dictionaries.
```

Suggesting we break this sentence up to make it a bit more digestible.
